test/e2e/storage/persistent_volumes-local.go

```diff
@@ -53,96 +54,238 @@ const (
+		})
 
-			By("Creating local PVC and PV")
-			createLocalPVCPV(config, testVol)
+		It("should be able to mount and write to the volume using one-command containers", func() {
+			By("Creating a pod to write to the PV")
+			writeCmd, readCmd := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+			writeThenReadCmd := fmt.Sprintf("%s;%s", writeCmd, readCmd)
+			podSpec := makeLocalPod(config, testVol, writeThenReadCmd)
+			f.TestContainerOutput("pod writes to PV", podSpec, 0, []string{testVol.hostDir})
+		})
+
+		It("should be able to mount volume and read from pod1", func() {
+			By("Creating pod1")
+			pod1, pod1Err := createLocalPod(config, testVol)
+			Expect(pod1Err).NotTo(HaveOccurred())
+
+			pod1NodeName, pod1NodeNameErr := podNodeName(config, pod1)
+			Expect(pod1NodeNameErr).NotTo(HaveOccurred())
+			framework.Logf("pod1 %q created on Node %q", pod1.Name, pod1NodeName)
+			Expect(pod1NodeName).To(Equal(node0.Name))
+
+			By("Reading in pod1")
+			//testFileContent was written during setupLocalVolume
+			_, readCmd := createWriteAndReadCmds(volumeDir, testFile, "" /*writeTestFileContent*/)
+			readOut := podRWCmdExec(pod1, readCmd)
+			Expect(readOut).To(ContainSubstring(testFileContent)) /*aka writeTestFileContents*/
 
-			By("Creating a pod to consume the PV")
-			readCmd := fmt.Sprintf("cat /mnt/volume1/%s", testFile)
-			podSpec := createLocalPod(config, testVol, readCmd)
-			f.TestContainerOutput("pod consumes PV", podSpec, 0, []string{testContents})
+			By("Deleting pod1")
+			framework.DeletePodOrFail(config.client, config.ns, pod1.Name)
 		})
 
-		It("should be able to mount and write to the volume", func() {
-			By("Initializing test volume")
-			testVol = setupLocalVolume(config, node)
+		It("should be able to mount volume and write from pod1", func() {
+			By("Creating pod1")
+			pod1, pod1Err := createLocalPod(config, testVol)
+			Expect(pod1Err).NotTo(HaveOccurred())
 
-			By("Creating local PVC and PV")
-			createLocalPVCPV(config, testVol)
+			pod1NodeName, pod1NodeNameErr := podNodeName(config, pod1)
+			Expect(pod1NodeNameErr).NotTo(HaveOccurred())
+			framework.Logf("pod1 %q created on Node %q", pod1.Name, pod1NodeName)
+			Expect(pod1NodeName).To(Equal(node0.Name))
 
-			By("Creating a pod to write to the PV")
-			testFilePath := filepath.Join("/mnt/volume1", testFile)
-			cmd := fmt.Sprintf("echo %s > %s; cat %s", testVol.hostDir, testFilePath, testFilePath)
-			podSpec := createLocalPod(config, testVol, cmd)
-			f.TestContainerOutput("pod writes to PV", podSpec, 0, []string{testVol.hostDir})
+			By("Writing in pod1")
+			writeCmd, _ := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+			podRWCmdExec(pod1, writeCmd)
+
+			By("Deleting pod1")
+			framework.DeletePodOrFail(config.client, config.ns, pod1.Name)
+		})
+	})
+
+	Context("when two pods request one prebound PVC one after other", func() {
+		var (
+			testVol *localTestVolume
```

Comment:
same nit as before. it may just be me. But single variables read better via - `var testVol *localTestVolume`